pkg/apis/monitoring/v1/servicemonitor_types.go

```diff
@@ -83,6 +83,16 @@ type ServiceMonitorSpec struct {
 
 	// Label selector to select the Kubernetes `Endpoints` objects to scrape metrics from.
 	Selector metav1.LabelSelector `json:"selector"`
+
+	// Mechanism used to select the endpoints to scrape.
```

Comment:
It requires Prometheus >= v2.17.0. It's very unlikely that people run such old versions but to be consistent, I'd advise to reject the resource if the role selector mechanism is defined and the version doesn't meet the requirement.